src/main/java/hudson/plugins/sonar/client/WsClient.java

```diff
@@ -35,34 +34,45 @@
   public static final String API_PROJECT_STATUS = "/api/qualitygates/project_status?projectKey=";
   public static final String API_VERSION = "/api/server/version";
   public static final String API_PROJECT_NAME = "/api/projects/index?format=json&key=";
+  public static final String API_CE_TASK = "/api/ce/task?id=";
 
   private final HttpClient client;
   private final String serverUrl;
   private final String username;
   private final String password;
-  private final String projectKey;
 
-  public WsClient(HttpClient client, String serverUrl, String projectKey, String username, String password) {
+  public WsClient(HttpClient client, String serverUrl, String username, String password) {
     this.client = client;
     this.serverUrl = serverUrl;
-    this.projectKey = projectKey;
     this.username = username;
     this.password = password;
   }
 
-  public ProjectQualityGate getQualityGate52() throws Exception {
+  public CETask getCETask(String taskId) throws Exception {
+    String url = serverUrl + API_CE_TASK + taskId;
+    String text = client.getHttp(url, username, password);
+    JSONObject json = (JSONObject) JSONSerializer.toJSON(text);
+    JSONObject task = json.getJSONObject("task");
+
+    String status = task.getString("status");
+    String componentName = task.getString("componentName");
+    String componentKey = task.getString("componentKey");
+
+    return new CETask(status, componentName, componentKey, url);
+  }
+
+  public ProjectQualityGate getQualityGate52(String projectKey) throws Exception {
     String url = serverUrl + API_PROJECT_STATUS + encode(projectKey);
     String text = client.getHttp(url, username, password);
     JSONObject json = (JSONObject) JSONSerializer.toJSON(text);
     JSONObject projectStatus = json.getJSONObject("projectStatus");
 
     String status = projectStatus.getString("status");
-    String projectName = getProjectName();
 
-    return new ProjectQualityGate(projectName, status);
+    return new ProjectQualityGate(null, status);
   }
 
-  public ProjectQualityGate getQualityGateBefore52() throws Exception {
+  public ProjectQualityGate getQualityGateBefore52(String projectKey) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00112)